src/porepy/numerics/linalg/matrix_operations.py

```diff
@@ -35,23 +40,34 @@ def zero_columns(A, cols):
     col_indptr = mcolon(indptr[cols], indptr[cols + 1])
     A.data[col_indptr] = 0
 
+    if diag is not None:
+        # now we set the diagonal
+        diag_vals = np.zeros(A.shape[1])
+        diag_vals[cols] = diag
+        A += sps.dia_matrix((diag_vals, 0), shape=A.shape)
+
 
-@pp.time_logger(sections=module_sections)
-def zero_rows(A, rows):
+def zero_rows(
+    A: sps.csr_matrix, rows: np.ndarray, diag: Optional[np.ndarray] = None
+) -> None:
```

Comment:
See previous comment about diag. Consider rename to diag_values? Also, using "diag" assumes A is square, right? All in all, do we really want this?

Reply:
Agree, but upon looking, I also found this is unused code, so I removed the option instead.